src/util/postgres_client.js

```diff
@@ -1284,6 +1310,33 @@ class PostgresClient extends EventEmitter {
             if (pg_client) await pg_client.end();
         }
     }
+
+    async upgradeSequence(name) {
```

Comment:
this should be part of the upgrade script. it will not be used in any other context

Reply:
Initially, I also thought so 😀 however after consideration I've started leaning towards the current logic code breakdown. 

We are dealing with a change in DB client model mapping into the relational DB, specifically how sequences are mapped and their transformation.  The present approach looks cleaner responsibility-wise. Both the Postgres client and the upgrade script look cleaner and tidier, with no code duplication.

Reply:
I agree. We should remove

Reply:
I prefer to have the upgrade script as self-contained as possible, with as few dependencies as possible.
Also, once the upgrade is completed, we are left with a dead code that has no other purpose. 

I suggest providing a function in the db client to perform direct SQL queries and use that for the upgrade script.

Reply:
I would prefer to reuse the existing code, instead of having  duplication in the implementation of:

- fetching current value from a collection-based sequence
- native Postgres sequence creation
- etc...

IMHO sequence upgrade is similar to DB [Creation](https://github.com/noobaa/noobaa-core/blob/44b8f84e0af51abe952d9c0a44891f3ad18ebe9f/src/util/postgres_client.js#L1287-L1299), [Drop](https://github.com/noobaa/noobaa-core/blob/44b8f84e0af51abe952d9c0a44891f3ad18ebe9f/src/util/postgres_client.js#L1287-L1299), collection [create](https://github.com/noobaa/noobaa-core/blob/44b8f84e0af51abe952d9c0a44891f3ad18ebe9f/src/util/postgres_client.js#L1410-L1423) or new sequence [create](https://github.com/baum/noobaa-core/blob/1e7f1aab2c3bbefc45fccb129b21eb8306bbb8bd/src/util/mongo_client.js#L577-L579): those are the responsibility of the corresponding DB Client.  The responsibility of the upgrade script is to [verify the DB Type](https://github.com/baum/noobaa-core/blob/danny-mdsequence-cherries/src/upgrade/upgrade_scripts/5.12.0/upgrade_postgres_native_sequence.js#L8-L14) and let the DB client do SQL or other DB-specific operations since the DB client is the layer that is responsible for those kind of stuff.  Finally, the client's upgrade code is self-contained, once the versions of the core which used the collection-based sequence are no longer supported, the Postgres client [upgrade function](https://github.com/baum/noobaa-core/blob/1e7f1aab2c3bbefc45fccb129b21eb8306bbb8bd/src/util/postgres_client.js#L1314-L1338) could be easily removed...

Reply:
Another thought is we could incorporate the logic of [if there is a table, convert to native sequence](https://github.com/baum/noobaa-core/blob/1e7f1aab2c3bbefc45fccb129b21eb8306bbb8bd/src/util/postgres_client.js#L1314-L1338) into [postgres_client.define_sequence()](https://github.com/baum/noobaa-core/blob/1e7f1aab2c3bbefc45fccb129b21eb8306bbb8bd/src/util/postgres_client.js#L1463-L1465) logic, and achieve lazy migration that way. This way we don't need the upgrade script at all.